FilePersistence/src/version_control/merge/ListMergeComponentV2.cpp

```diff
@@ -205,4 +206,58 @@ void ListMergeComponentV2::computeOffsetsInBranch(const QList<Model::NodeIdType>
 	}
 }
 
+void ListMergeComponentV2::removeHoles(const QList<Model::NodeIdType> lists, GenericTree* tree, ChangeGraph& cg)
+{
+	for (auto listId : lists)
+	{
+		// Traverse through children labels
+		ChangeGraph::IdToLabelMap map;
+		int i = 0;
+		int label = 0;
+		int offset = 1;
+		auto listNode = tree->find(listId, true);
+		auto idList = nodeListToSortedIdList(listNode->children());
+		QHash< QString, QPair<int, int>> labelsToBeUpdated;
+		while (i < idList.size())
+		{
+			auto node = listNode->child(QString::number(label));
+			if (node)
+			{
+				map.insert(idList.at(i), {QString::number(i), MergeChange::Branch::None});
+				i++;
+				offset = 1;
+			}
+			else
+				labelsToBeUpdated.insert(QString::number(label), qMakePair(i-1, offset++));
+			label++;
+		}
+		// Holes at the end of listy are not added in the lableToBeUpdated
+
+		auto listChanges = cg.changesForChildren(listId);
+		for (auto change : listChanges)
+		{
+			// No need to update if change is MoveOut/Deletion Or Stationary change(Not Label)
+			if (change->type() == ChangeType::Deletion ||
+				(change->type() == ChangeType::Move && change->oldParentId() == listId) ||
+				 (change->type() == ChangeType::Stationary && !change->updateFlags().testFlag(ChangeDescription::Label)))
+					continue;
+
+			// Insertion Or MoveIn or LabelChange
+
+			// If change corresponding to the label is not present, then hole must be at the end
+			QPair<int, int> fractionalIndex;
+			auto fractionalIndexIt = labelsToBeUpdated.constFind(change->newLabel());
+			if (fractionalIndexIt == labelsToBeUpdated.constEnd())
+				fractionalIndex = qMakePair(idList.size()-1, offset++);
+			else
+				fractionalIndex = labelsToBeUpdated.constFind(change->newLabel()).value();
```

Comment:
Here you can just use `fractionalIndexIt.value()`. This is why I mentioned before that you should assing that result to a variable, so that you can reuse it here.